src/main.rs

```diff
@@ -148,28 +151,36 @@ pub(crate) extern "C" fn main(hart_id: usize, device_tree_blob_addr: usize) -> !
 }
 
 fn main_loop(ctx: &mut VirtContext, mctx: &mut MiralisContext) -> ! {
-    let mut bench = Benchmark::new();
-
     loop {
         unsafe {
             if config::BENCHMARK {
-                bench.start();
+                let mut bench = BENCH.lock();
+                bench.start_interval_counter(IntervalCounter::ExecutionTime);
+                bench.start_interval_counter(IntervalCounter::InstructionRet);
```

Comment:
Do we want to manually call the start and stop multiple time or could the function start/stop all enabled counters at once?

Reply:
It can be all at once sure, I'll remove the all or nothing.